cloud-agnostic/core/src/NamedDependency.ts

```diff
@@ -0,0 +1,43 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import { Container, interfaces } from "inversify";
+
+import { Dependency } from "./Dependency";
+import { DependencyConfig } from "./DependencyConfig";
+
+export abstract class NamedDependency extends Dependency {
+  protected abstract bindInstances(
```

Comment:
This name is also confusing. You're binding types for a single named instance, not for multiple named instances.

Reply:
Fixed